interpreter/cling/lib/Interpreter/DeclCollector.cpp

```diff
@@ -57,6 +58,15 @@ namespace {
         return true;
     return false;
   }
+
+  /// \brief Asserts that the given transaction is not null, otherwise prints a
+  /// stack trace to stderr.
+  static void assertHasTransaction(const cling::Transaction* transaction) {
+    if (transaction == nullptr) {
+      llvm::sys::PrintStackTrace(llvm::errs());
+      assert(false && "Missing transaction during deserialization!");
```

Comment:
Shouldn't we use llvm_unreachable here. If so we will need to rename the routine.

Reply:
The original code was using `assert`. For me this looks like a poster child of `assert`: validating a precondition is met.

Reply:
According to some [rules](http://llvm.org/docs/CodingStandards.html#assert-liberally) `llvm_unreachable` is the preferred primitive. I do not have a strong opinion here, but the closer to the llvm coding guidelines we are in cling, the better.

Reply:
Good points, Vassil. Thanks for explaining!